ygen/enumgen.go

```diff
@@ -440,103 +562,132 @@ func findEnumSet(entries map[string]*yang.Entry, compressPaths, noUnderscores, s
 type enumGenState struct {
 	// definedEnums keeps track of generated enum names to avoid conflicts.
 	definedEnums map[string]bool
-	enumSet      *enumSet
+	// enumSet contains the final, deduped generated enumerated value names
+	// to be returned by findEnumSet.
+	enumSet *enumSet
+	// enumeratedTypedefDedupSets stores the set of unique string keys
+	// representing typedef enumeration usages that were mapped to the same
+	// initial generated name using the basic name generation rules.
+	// This is an intermediate storage for these names. While the initial
+	// generated name is ideally stored inside enumSet directly,
+	// enumeration names may collide, and thus need de-duplication before
+	// they're stored inside enumSet. This is the reason for being for all
+	// dedup sets within enumGenState.
+	enumeratedTypedefDedupSets map[string]map[string]bool
+	// enumeratedLeafDedupSets stores the unique string keys representing
+	// enumeration leaves that were mapped to the same initial generated
+	// name using the basic name generation rules.
+	enumeratedLeafDedupSets map[string]map[string]bool
```

Comment:
Can there ever be cases where a key in `enumeratedTypedefDedupSets` is the same as a key in `enumeratedLeafDedupSets`?

Reply:
No, but the way their name clashes are resolved are different, so the different maps serve to separate their processing.